### DISCUSSION TITLE
How to show only part of the content when Transclusing

### DISCUSSION THREAD

**Discussion by dongrentianyu** [TiddlyWiki/TiddlyWiki5#7055] [Ideas]
How to show only part of the content when Transclusing

I thought of a new plugin that I would like to develop. The requirement scenario is something like this "I sometimes need to quote from an article quite often. Sometimes I only need to quote the first few sentences, sometimes I need to quote the last few sentences, and sometimes I need to quote all of it. The whole thing is fine, just embed it, but other times I only need part of it and that's difficult. Like a long poem by Du Fu '

Song of a thatched cottage broken by the autumn wind

Dynasty: Tang Dynasty|Author: Du Fu

In August, the wind blows furiously and rolls up my house with a triple thatch. The thatch flew across the river and spilled onto the outskirts of the river; the higher ones hung from the top of the long forest, the lower ones drifted around the sunken pond.

The children in the southern village deceived me for being old and feeble, so I could not bear to be a thief across the river. I went into the bamboo with my arms, but my lips were burning and I could not call out, so I leaned on my staff and sighed.

The wind soon settled and the clouds were inky; autumn was indifferent to the darkness. The cloth bedding has been as cold as iron for many years, and the child has been lying on the ground. There is no dryness in the leaks at the head of the bed, and the rain is as heavy as a flax. I have not slept much since the tragedy, how can I get through the long nights?

I have been able to secure a house with ten million rooms, to shelter the world's cold people, to keep the wind and rain at bay. Oh, my! When will I see this house before my eyes, it will be enough for me to die of cold in my house alone!

Sometimes I may only need to quote part of it. It wouldn't feel right if I broke the poem up into many smaller entries."

My idea was to hide it via css. Firstly, work out how many strings are needed to show the content, followed by how many strings follow. If I wanted to show this paragraph 

'In August the wind blows furiously and rolls up my house with three strings of grass. The grass flew across the river and spilled onto the outskirts of the river; the taller ones hung in the tops of the long forests, the lower ones floated around the sunken ponds.'

 Suppose it is at the 123rd string, followed by the 234th string at the end. Then set the string to be displayed except for the 123rd to 234th strings and nothing else. This will do the job without slicing the entries and eliminating the copying operation. And you can use this the next time you want to show another fragment.

For example, I wrote this in a writing scene "

Du Fu writes in [this poem|Song of a thatched cottage broken by the autumn wind]

{{Song of a thatched cottage broken by the autumn wind}(123,234)}

expressing his noble sentiment of thinking about the world.

Or there is another way, the other way around, where the string from the first to the last is not shown and the others are shown.

But there's the problem that, for example, I didn't cut up the poem Du Fu at the time, so the order of the strings was 123rd to 234th, but later I cut up the poem, and that order changed. It might then become 23rd to 113th. That's a bit of a problem.

The idea is to avoid cutting the entries too small or duplicating them, and to avoid copying them too often, but at the moment it might require the use of the relink plugin code or implementation ideas. And as I am not a programmer and do not program development, any kind of discussion is welcome.

**Comment by dongrentianyu** [comment]
I was thinking of using a different example to illustrate the above, or choosing a long poem in English, but I have not studied any form of English poetry, so I have had to use a poem written by the great Chinese poet Du Fu, I hope you can understand.

**Comment by dongrentianyu** [comment]
Also, I don't speak English, and everything above is machine translated, but I will pay attention to any discussion you may have.

**Comment by XiaoYun** [comment]
呵呵！收到，待会查看你的邮件。

**Comment by AnthonyMuscio** [comment]
I believe there are ways to achieve this in wikitext and html 

Perhaps raise this over in talk.tiddlywiki.org

You could wrap some text in a html section or arbitrary tag, then write a wiki macro that parses the tiddler and extracts everything between the named start and end tag.

```
text
<section1>
text 2
</section1>
text 3
```
In addition to this I have an editor toolbar button to select some text and wrap in in the tags. Quick and easy.

Then in the other tiddler you would use the macro

```
text a
<<macroname tiddlername section1>>
text b
```

Resulting in;
**text a
text 2
text b**

**Reply by dongrentianyu** [reply]
Thank you very much for your reply. I found out later that there is a plugin that has solved my problem. But I forgot to close this issues, sorry. Here is the URL.

https://kookma.github.io/TW-Hsl/

Thanks again for your kind reply!

**Reply by dongrentianyu** [reply]
Probably can't turn it off here, so it will have to stay on.

**Reply by dongrentianyu** [reply]
I've been thinking about this again recently, and I've found that the hsl plugin doesn't quite meet my needs, but I can't fix that by modifying the plugin.

I came across this issue again and I found that your solution might be very good. Use htmls tags for interception, but the problem is how to handle it if you want to call two or more parts of an entry repeatedly?

Use multiple <section1><section2><section3><section4>? That seems to work just fine. Can you give the content of the macro though? Thanks.

**Comment by XiaoYun** [comment]
呵呵！收到，待会查看你的邮件。
